packages/aws-cdk/lib/toolkit/error.ts

```diff
@@ -0,0 +1,48 @@
+const TOOLKIT_ERROR_SYMBOL = Symbol.for('@aws-cdk/core.TooklitError');
+const AUTHENTICATION_ERROR_SYMBOL = Symbol.for('@aws-cdk/core.AuthenticationError');
+
+/**
+ * Represents a general toolkit error in the AWS CDK Toolkit.
+ */
+class ToolkitError extends Error {
+  /**
+   * Determines if a given error is an instance of ToolkitError.
+   */
+  public static isToolkitError(x: any): x is ToolkitError {
+    return x !== null && typeof(x) === 'object' && TOOLKIT_ERROR_SYMBOL in x;
+  }
+
+  /**
+   * Determines if a given error is an instance of AuthenticationError.
+   */
+  public static isAuthenticationError(x: any): x is AuthenticationError {
+    return this.isToolkitError(x) && AUTHENTICATION_ERROR_SYMBOL in x;
+  }
+
+  /**
+   * The type of the error, defaults to "toolkit".
+   */
+  public readonly type: string;
+
+  constructor(message: string, type: string = 'toolkit') {
```

Comment:
should we restrict `type` to either `toolkit` or `authentication`? what was the decision making behind typing it as `string`?

Reply:
also, should the constructor be private?

Reply:
Both `ConstructError` and `ValidationError` have public constructors: https://github.com/aws/aws-cdk/blob/8718767c5bf9fce33ea80e6c7d3bb136c79dbf7d/packages/aws-cdk-lib/core/lib/errors.ts#L140

I could make it an enum with string values? But I'm pretty sure Momo wanted a type property that is a string (third bullet point here): https://github.com/aws/aws-cdk/issues/32347

Reply:
for all of these questions its my hope that you answer them with your reasons for why you've done something, not because Momo said to or because a prior implementation did so. Both of those things can be true, but you should at least know why you're doing it. 

now for this case my question was a silly one. you have a public constructor because you are calling `new ToolkitError` everywhere